yaksh/models.py

```diff
@@ -134,6 +140,346 @@ def create_trial_course(self, user):
+    # complete.
+    active = models.BooleanField(default=True)
+
+    # Description of quiz.
+    description = models.CharField(max_length=256)
+
+    # Mininum passing percentage condition.
+    pass_criteria = models.FloatField("Passing percentage", default=40)
+
+    # Number of attempts for the quiz
+    attempts_allowed = models.IntegerField(default=1, choices=attempts)
+
+    time_between_attempts = models.IntegerField(
+        "Number of Days", choices=days_between_attempts
+    )
+
+    is_trial = models.BooleanField(default=False)
+
+    instructions = models.TextField('Instructions for Students',
+                                    default=None, blank=True, null=True)
+
+    view_answerpaper = models.BooleanField('Allow student to view their answer\
+                                            paper', default=False)
+
+    allow_skip = models.BooleanField("Allow students to skip questions",
+                                     default=True)
+
+    weightage = models.FloatField(default=1.0)
+
+    creator = models.ForeignKey(User, null=True)
+
+    objects = QuizManager()
+
+    class Meta:
+        verbose_name_plural = "Quizzes"
+
+    def is_expired(self):
+        return not self.start_date_time <= timezone.now() < self.end_date_time
+
+    def create_demo_quiz(self, user):
+        demo_quiz = Quiz.objects.create(
+            start_date_time=timezone.now(),
+            end_date_time=timezone.now() + timedelta(176590),
+            duration=30, active=True,
+            attempts_allowed=-1, time_between_attempts=0,
+            description='Yaksh Demo quiz', pass_criteria=0,
+            creator=user
+        )
+        return demo_quiz
+
+    def get_total_students(self, course):
+        return AnswerPaper.objects.filter(
+                question_paper=self.questionpaper_set.get().id,
+                course=course
+            ).values_list("user", flat=True).distinct().count()
+
+    def get_passed_students(self, course):
+        return AnswerPaper.objects.filter(
+                question_paper=self.questionpaper_set.get().id,
+                course=course, passed=True
+            ).values_list("user", flat=True).distinct().count()
+
+    def get_failed_students(self, course):
+        return AnswerPaper.objects.filter(
+                question_paper=self.questionpaper_set.get().id,
+                course=course, passed=False
+            ).values_list("user", flat=True).distinct().count()
+
+    def __str__(self):
+        desc = self.description or 'Quiz'
+        return '%s: on %s for %d minutes' % (desc, self.start_date_time,
+                                             self.duration)
+
+
+##########################################################################
+class LearningUnit(models.Model):
+    """ Maintain order of lesson and quiz added in the course """
+    order = models.IntegerField()
+    learning_type = models.CharField(max_length=16)
```

Comment:
perhaps `type` is better than `learning_type`, also this should be a choice and not an arbitrary charfield.

Reply:
The user does not specifically create learning unit, it is our internal model. So learning type can be saved while adding quizzes and lesson to a module. So we don't need a choice field.